src/agent/src/device.rs

```diff
@@ -438,204 +459,174 @@ fn scan_scsi_bus(scsi_addr: &str) -> Result<()> {
 
 // update_spec_device updates the device list in the OCI spec to make
 // it include details appropriate for the VM, instead of the host.  It
-// is given the host path to the device (to locate the device in the
-// original OCI spec) and the VM path which it uses to determine the
-// VM major/minor numbers, and the final path with which to present
-// the device in the (inner) container
+// is given the old path to the device (as in the original OCI spec)
+// and the VM path which it uses to determine the VM major/minor
+// numbers, and the new_path which will be used in the final spec.
 #[instrument]
-fn update_spec_device(
+fn update_spec_devices(
     spec: &mut Spec,
-    devidx: &DevIndex,
-    host_path: &str,
-    vm_path: &str,
-    final_path: &str,
+    fixups: &[DevSpecFixup],
 ) -> Result<()> {
-    let major_id: c_uint;
-    let minor_id: c_uint;
-
-    // If no container_path is provided, we won't be able to match and
-    // update the device in the OCI spec device list. This is an error.
-    if host_path.is_empty() {
-        return Err(anyhow!("Host path cannot empty for device"));
-    }
-
-    let linux = spec
-        .linux
-        .as_mut()
-        .ok_or_else(|| anyhow!("Spec didn't container linux field"))?;
+    let devidx = DevIndex::new(spec);
 
-    if !Path::new(vm_path).exists() {
-        return Err(anyhow!("vm_path:{} doesn't exist", vm_path));
-    }
+    for fixup in fixups {
+        let major_id: c_uint;
+        let minor_id: c_uint;
 
-    let meta = fs::metadata(vm_path)?;
-    let dev_id = meta.rdev();
-    unsafe {
-        major_id = major(dev_id);
-        minor_id = minor(dev_id);
-    }
+        // If no container_path is provided, we won't be able to match and
+        // update the device in the OCI spec device list. This is an error.
+        if fixup.old.is_empty() {
+            return Err(anyhow!("Host path cannot empty for device"));
+        }
 
-    info!(
-        sl!(),
-        "update_spec_device(): vm_path={}, major: {}, minor: {}\n", vm_path, major_id, minor_id
-    );
+        let linux = spec
+            .linux
+            .as_mut()
+            .ok_or_else(|| anyhow!("Spec didn't container linux field"))?;
```

Comment:
It could be, but it's out of scope for the patches so far.  The intention is that the current patch just re-indent the existing update_device_spec() code with a new outer block, without rearranging, for ease of review.  I'll be making further rearrangements in later patches.